### DISCUSSION TITLE
Support for batch migrate and rollback operations?

### DISCUSSION THREAD

**Discussion by raydwaipayan** [amacneil/dbmate#359] [Feature Ideas]
It will be really nice if there is a support to migrate and rollback in batches.
Generally when releases might contain multiple migrations, it should be
possible to group these into batches and then refer by the batch name.

Something like:

```
dbmate batch migrate # Migrate to latest
dbmate batch migrate <batch_name> # Run all migrations upto <batch_name>

dbmate batch rollback # Rollback latest batch
dbmate batch rollback <batch_name> # Rollback all migrations down to <batch_name>
```

Thoughts?

**Comment by amacneil** [comment]
How are you envisaging the batch is declared?

I think this could be partially solved by https://github.com/amacneil/dbmate/issues/18 (migrating or rolling back to a specific migration).

I'm not sure of the context for your release (e.g. whether you are publishing something that customers would individually run migrations), but an approach I would consider is squashing your migrations prior to a release into a single file. You could still write each migration in an idempotent manner (`create table if exists`, etc) so that it doesn't conflict with existing state for your development team.

**Comment by gregwebs** [comment]
This seems the same as issue #243 

I suggest being able to specify specific migrations with `--migration=csv` and adding a `--batch-transaction` flag.

**Comment by gregwebs** [comment]
PR #438 adds a syntax for multiple migrations (or maybe you just wanted all unapplied in which case you wouldn't need that). Would need an additional flag for batching.
